homeassistant/components/bluetooth_tracker/device_tracker.py

```diff
@@ -60,13 +60,18 @@ def is_bluetooth_device(device: Device) -> bool:
 
 def discover_devices(device_id: int) -> list[tuple[str, str]]:
     """Discover Bluetooth devices."""
-    result = bluetooth.discover_devices(
-        duration=8,
-        lookup_names=True,
-        flush_cache=True,
-        lookup_class=False,
-        device_id=device_id,
-    )
+    try:
+        result = bluetooth.discover_devices(
+            duration=8,
+            lookup_names=True,
+            flush_cache=True,
+            lookup_class=False,
+            device_id=device_id,
+        )
+    except OSError as ex:
+        # OSError is generally thrown if a bluetooth device isn't found
+        _LOGGER.error("Couldn't discover bluetooth devices: %s", ex)
```

Comment:
If it's a comment error, we should not report it in log as an error. More like debug.

Reply:
Not sure what you mean by "comment error".
I suggest users do want/need an error to explain why the Bluetooth stack isn't functioning - just not an uncaught exception and full stack trace as is happening now (see linked issue for current behaviour).

Reply:
Comment -> common. Failing to connect to a device seem like it should not be logged at error level. We are talking about a tracker here. It's perfectly normal to fail to connect to a device if it's out of range. It's not an error.

Reply:
It's not that it can't find an endpoint, it's that a Bluetooth adapter can't be found (usb, pcie, etc.) connected to the host. This means that the tracker can't function to search for endpoints.